### PR TITLE
[BFLY-790] New Contracts Integration

### PR DESCRIPTION
**Pull request by malikkulsoom**
- added auth layer
- updated library
- contract changes
- updated adjust leverage
- added adjust margin function

**Tasks**:
https://linear.app/seed/issue/BFLY-790/update-contract-calls-in-firefly-client

### CODE REVIEW

File: src/fireflyClient.ts [hunk 1]
```diff
@@ -350,8 +342,8 @@ export class FireflyClient {
    * @returns margin balance of positions of given symbol
    */
   async getAccountPositionBalance(symbol: MarketSymbol, perpContract?: address) {
-    const perpV1Contract = this.getContract("PerpetualProxy", perpContract, symbol);
-    const marginBalance = await perpV1Contract.connect(this.getWallet()).getAccountPositionBalance(this.getPublicAddress());
+    const perpV1Contract = this.getContract(this._perpetual, perpContract, symbol);
+    const marginBalance = await perpV1Contract.connect(this.getWallet()).getAccountBalance(this.getPublicAddress());
```
Comment (YameenMalik) [inline src/fireflyClient.ts:346]: the variable name `marginBalance` is misleading as it implies you are getting balance of user in margin bank, but you actually are getting user's position balance, best to remove this variable completely and just write:

`return await perpV1Contract.connect(this.getWallet()).getAccountBalance(this.getPublicAddress());`

File: src/fireflyClient.ts [hunk 2]
```diff
@@ -233,7 +233,7 @@ export class FireflyClient {
    * @returns Boolean true if user is funded, false otherwise
    */
   async mintTestUSDC(contract?: address): Promise<boolean> {
-    const tokenContract = this.getContract("USDTToken", contract);
+    const tokenContract = this.getContract(this._usdcToken, contract);
```
Comment (YameenMalik) [inline src/fireflyClient.ts:236]: Can we check over here that what is the network user is connected to? If it is dev or testnet, we should not mint test tokens.

File: src/fireflyClient.ts [hunk 3]
```diff
@@ -644,6 +592,47 @@ export class FireflyClient {
...
+  * Add or remove margin from the open position
+  * @param symbol market symbol of the open position
+  * @param operationType operation you want to perform `Add` | `Remove` margin
+  * @param amount (number) amount user wants to add or remove from the position
+  * @param perpetualAddress (address) address of Perpetual contract
+  * @returns boolean value indicating if margin adjusted successfully
+  */
+ async adjustMargin(
+  symbol: MarketSymbol,
+  operationType: "Add" | "Remove",
```
Comment (YameenMalik) [inline src/fireflyClient.ts:605]: better to make an enum called ADJUST_MARGIN having two values Add and Remove.

**Review by YameenMalik — CHANGES_REQUESTED** [review]
Looks good. Entertain small comments before I can merge.

File: src/fireflyClient.ts (hunk 2 above)
Comment (malikkulsoom) [reply src/fireflyClient.ts:236]: But on the function description above, we have written that it works on testnet only. Let me know on which environments it should work and i'll update the docs too

Comment (YameenMalik) [reply src/fireflyClient.ts:236]: this function will also work on devnet but I don't think normal user's will have access to that, only developers will know about it. Either you can check at the beginning of the function what network user is connected to, if its PRODUCTION, this function should throw an error.

Comment (malikkulsoom) [reply src/fireflyClient.ts:236]: Network is the object containing RPC url, chain id etc, Currently, we only have two added TESTNET and DEV, so I'll just create a condition if Network is other than any of these two throw error

**Review by YameenMalik — APPROVED** [review]

**Pull request merged**